nibabel/cifti2/cifti2.py

```diff
@@ -1484,8 +1485,14 @@ def update_headers(self):
         >>> img.update_headers()
         >>> img.nifti_header.get_data_shape() == (2, 3, 4)
         True
+        >>> np.array_equal(img.nifti_header['dim'], [7, 1, 1, 1, 1, 2, 3, 4])
+        True
         """
         self._nifti_header.set_data_shape(self._dataobj.shape)
+        _dims = np.ones((8), dtype=int)
+        _dims[0] = 7 if len(self._dataobj.shape) == 3 else 6
+        _dims[5:8] = (self._dataobj.shape + (1,))[:3]
+        self._nifti_header['dim'] = _dims
```

Comment:
Updating the input to `set_data_shape()` ought to do the trick.

```suggestion
        self._nifti_header.set_data_shape((1, 1, 1, 1) + self._dataobj.shape)
```

Reply:
this isn't equivalent though? `dim`[0] should be either 6 or 7

Reply:
`dim0` indicates the length of `dim[1:8]`. This wlll be determined by this tuple.